src/main/java/org/folio/oaipmh/helpers/AbstractHelper.java

```diff
@@ -315,6 +354,25 @@ private String getUntilDate(Request request, String from) {
     }
   }
 
+  protected boolean filterInstance(Request request, JsonObject instance){
+    if (!isDeletedRecordsEnabled(request, REPOSITORY_DELETED_RECORDS)){
+      return !storageHelper.isRecordMarkAsDeleted(instance);
+    } else {
+      Map<String, String> okapiHeaders = request.getOkapiHeaders();
+      return getBooleanProperty(okapiHeaders, REPOSITORY_SUPPRESSED_RECORDS_PROCESSING)
```

Comment:
Could you create please the boolean variable here "shouldProcessSuppressedRecords" and assign to it value of the setting and use such variable instead of "getBooleanProperty (okapiHeaders, REPOSITORY_SUPPRESSED_RECORDS_PROCESSING)" which takes a lot of space and make code less readable.